plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergParquetPageSkipping.java

```diff
@@ -0,0 +1,139 @@
+import io.trino.testing.QueryRunner;
+import org.intellij.lang.annotations.Language;
+import org.testng.SkipException;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Map;
+
+import static com.google.common.io.Resources.getResource;
+import static io.trino.testing.sql.TestTable.randomTableSuffix;
+import static java.lang.String.format;
+import static java.nio.file.StandardCopyOption.REPLACE_EXISTING;
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class TestIcebergParquetPageSkipping
+        extends BaseTestParquetPageSkipping
+{
+    @Override
+    protected QueryRunner createQueryRunner()
+            throws Exception
+    {
+        return IcebergQueryRunner.builder()
+                .setIcebergProperties(Map.of(
+                        "iceberg.file-format", "parquet",
+                        // testAndPredicates method replaces Parquet file directly
+                        "iceberg.use-file-size-from-metadata", "false",
+                        "parquet.use-column-index", "true",
+                        // Small max-buffer-size allows testing mix of small and large ranges in HdfsParquetDataSource#planRead
+                        "parquet.max-buffer-size", "400B"))
+                .build();
+    }
+
+    @Override
+    protected String tableDefinitionForSortedTables(String tableName, String sortByColumnName, String sortByColumnType)
+    {
+        if (sortByColumnType.equals("smallint")) {
+            throw new SkipException("Unsupported column type: " + sortByColumnType);
+        }
+        if (sortByColumnType.equals("timestamp")) {
+            sortByColumnType = "timestamp(6)";
+        }
+        String createTableTemplate =
+                "CREATE TABLE %s ( " +
+                        "   orderkey bigint, " +
+                        "   custkey bigint, " +
+                        "   orderstatus varchar(1), " +
+                        "   totalprice double, " +
+                        "   orderdate date, " +
+                        "   orderpriority varchar(15), " +
+                        "   clerk varchar(15), " +
+                        "   shippriority integer, " +
+                        "   comment varchar(79), " +
+                        "   rvalues double array " +
+                        ") " +
+                        "WITH ( " +
+                        "   format = 'PARQUET' " +
+                        ")";
+        createTableTemplate = createTableTemplate.replaceFirst(sortByColumnName + "[ ]+([^,]*)", sortByColumnName + " " + sortByColumnType);
+        return format(createTableTemplate, tableName, sortByColumnName);
+    }
+
+    @Override
+    public void testAndPredicates()
+            throws Exception
+    {
+        // Override because optimized Parquet writer used in Iceberg doesn't support column index at this time
```

Comment:
I wonder if we should wait to have native writer support for page indexes before we implement this.
Otherwise users don't have a straightforward way to make use of this functionality.

Reply:
Yes, the current default parquet writer in hive connector writes them, although that would stop once we change the default to native parquet writer (unless we also implement it there).
While that is a better way of testing, it's still difficult for end users to benefit from this feature.

Reply:
It is not ideal, but testing the feature is a bit of a chicken/egg problem unless we put both in at the same time.

Reply:
I don't mind suspending this PR until native Parquet writer support page indexes.